File: backend/Backend/Message.cs
namespace Backend;

public class Message
{
    public Guid Id { get; set; }
    public Guid RoomId { get; set; }
    public string Content { get; set; }
    public DateTime TimeSent { get; set; }
    public Guid UserId { get; set; }

    public Message(Guid roomId, Guid id, string content, DateTime timeSent, Guid userId)
    {
        RoomId = roomId;
        Id = id;
        Content = content;
        TimeSent = timeSent;
        UserId = userId;
    }
}

File: frontend/lib/Message.dart
class Message {
  final String id;
  final String roomId;
  final String content;
  final DateTime timeSent;
  final String userId;
  String username;

  Message({
    required this.id,
    required this.roomId,
    required this.content,
    required this.timeSent,
    required this.userId,
    required this.username
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id'],
      roomId: json['roomId'],
      content: json['content'],
      timeSent: DateTime.parse(json['timestamp']),
      userId: json['senderId'],
      username: json["username"]
    );
  }
}
